File: js/src/utils/getPythFeedAccountKey.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPythFeedAccountKey = void 0;
const buffer_1 = require("buffer");
const web3_js_1 = require("@solana/web3.js");
const constants_1 = require("../constants");
const getPythFeedAccountKey = (shard, priceFeed) => {
    const buffer = buffer_1.Buffer.alloc(2);
    buffer.writeUint16LE(shard);
    return web3_js_1.PublicKey.findProgramAddressSync([buffer, buffer_1.Buffer.from(priceFeed)], constants_1.DEFAULT_PYTH_PUSH_PROGRAM);
};
exports.getPythFeedAccountKey = getPythFeedAccountKey;

File: python/src/utils/get_pyth_feed_account_key.py
from typing import Tuple, List

from solders.pubkey import Pubkey

from constants import DEFAULT_PYTH_PUSH_PROGRAM


def get_pyth_feed_account_key(shard: int, price_feed: List[int]) -> Tuple[Pubkey, int]:
    buffer = bytearray(2)
    buffer[0:2] = shard.to_bytes(2, "little")
    return Pubkey.find_program_address(
        [bytes(buffer), bytes(price_feed)],
        DEFAULT_PYTH_PUSH_PROGRAM,
    )
